src/browser-polyfill.js

```diff
@@ -191,33 +205,46 @@ if (typeof browser === "undefined" || Object.getPrototypeOf(browser) !== Object.
       });
     };
 
+    /**
+     * Determines whether an object has a property with the specified name.
+     *
+     * @param {any} target The object to test.
+     * @param {string | number | symbol} v A property name.
+     *
+     * @type {function(any, string | number | symbol):boolean}
+     */
     let hasOwnProperty = Function.call.bind(Object.prototype.hasOwnProperty);
 
     /**
      * Wraps an object in a Proxy which intercepts and wraps certain methods
      * based on the given `wrappers` and `metadata` objects.
      *
-     * @param {object} target
+     * @template T
+     *
+     * @param {T} target
      *        The target object to wrap.
      *
-     * @param {object} [wrappers = {}]
+     * @param {object} [wrappers]
      *        An object tree containing wrapper functions for special cases. Any
      *        function present in this object tree is called in place of the
      *        method in the same location in the `target` object tree. These
-     *        wrapper methods are invoked as described in {@see wrapMethod}.
+     *        wrapper methods are invoked as described in {@link wrapMethod}.
      *
-     * @param {object} [metadata = {}]
+     * @param {object} [metadata]
      *        An object tree containing metadata used to automatically generate
      *        Promise-based wrapper functions for asynchronous. Any function in
      *        the `target` object tree which has a corresponding metadata object
      *        in the same location in the `metadata` tree is replaced with an
      *        automatically-generated wrapper function, as described in
-     *        {@see wrapAsyncFunction}
+     *        {@link wrapAsyncFunction}
      *
-     * @returns {Proxy<object>}
```

Comment:
The result of&nbsp;the&nbsp;`Proxy` constructor doesn’t work this way.

Reply:
What do you mean by that? Proxies have no prototype, Proxy instances are seemingly transparant wrappers around the original target object. Hence for human readers `Proxy<object>` should be clear enough. Please undo the changes in this whole comment (except maybe for `= {}`).

Reply:
I mean this:
https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es2015.proxy.d.ts#L40